### DISCUSSION TITLE
Using Injected, okxwallet or okexwallet can't connect with OKX wallet

### DISCUSSION THREAD

**Discussion by eureka0928** [wevm/wagmi#3811] [Q&A]
### Describe the bug

export const okxConnector = injected({target: 'okxWallet'})

using targeted okx wallet, either okxWallet or oKExWallet is not available to connect with the wallet and return ProviderNotFoundError, while using injected({target: 'metamask'}) is ok to connect with my metamask wallet

### Wagmi Version

2.5.19

### Viem Version

2.9.13

### TypeScript Version

5.2.2

### Check existing issues

- [ ] I checked there isn't [already an issue](https://github.com/wevm/wagmi/issues) for the bug I encountered.

**Comment by glitch-txs** [comment]
OKX supports EIP-6963, you can filter the connector by its id.

**Comment by eureka0928** [comment]
@glitch-txs , could you give me a advice to implement it correctly?
I used injected, but didn't work

This was what I did

```
injected({
      target() {
        return {
          id: "okxwallet",
          name: "OKX Wallet",
          provider: () => {
            if (typeof window !== "undefined") return

            const isOkxWallet = (ethereum) => {
              return !!ethereum?.isOkxWallet
            }

            if (isOkxWallet(window.ethereum)) {
              return window.ethereum
            }

            if (window.ethereum?.providers)
              return window.ethereum.providers.find(isOkxWallet) ?? null

            return window["okxwallet"] ?? null
          }
        }
      }
    })
```

**Reply by glitch-txs** [reply]
you don't need to do anything of this, Wagmi will create automatically the OKX connector if that wallet is installed in your browser, in the connector you'll have also access to the OKX wallet's icon

**Reply by glitch-txs** [reply]
check [EIP-6963](https://eips.ethereum.org/EIPS/eip-6963)

**Reply by eureka0928** [reply]
I made the custom button to connect, hmm, it won't work like coinbase wallet in Wagmi?

**Reply by glitch-txs** [reply]
If you create a small repo I can help you setting it up

**Reply by eureka0928** [reply]
https://github.com/leonace924/wagmi-connect-wallets 
@glitch-txs , I created just simple repo without styling, you can have a look and let me know
Thanks

**Reply by glitch-txs** [reply]
try searching for this id: `com.okex.wallet` you can remove the injected config

**Comment by eureka0928** [comment]
@jxom , would you mind to help me on this?
I didn't find the perfect code for OKX wallet and Binance
Thank you
